tests/test_common_gpu.py

```diff
@@ -274,3 +280,45 @@ def test_adaption_prompt_4bit(self):
 
         random_input = torch.LongTensor([[1, 0, 1, 0, 1, 0]]).to(0)
         _ = model(random_input)
+
+    @require_torch_gpu
+    @pytest.mark.single_gpu_tests
+    @require_bitsandbytes
+    def test_print_4bit_expected(self):
+        model = AutoModelForCausalLM.from_pretrained(
+            "facebook/opt-125m",
+            load_in_4bit=True,
+        )
+
+        config = LoraConfig(
+            r=8,
+        )
+        model = get_peft_model(model, config)
+        model.print_trainable_parameters()
+
+        captured = self.capsys.readouterr()
+        self.assertEqual(
+            "trainable params: 294,912 || all params: 125,534,208 || trainable%: 0.23492560689115113\n", captured.out
+        )
+
+        # test with double quant
+        bnb_config = BitsAndBytesConfig(
+            load_in_4bit=True,
+            bnb_4bit_use_double_quant=True,
+        )
+
+        model = AutoModelForCausalLM.from_pretrained(
+            "facebook/opt-125m",
+            quantization_config=bnb_config,
+        )
+
+        config = LoraConfig(
+            r=8,
+        )
+        model = get_peft_model(model, config)
+        model.print_trainable_parameters()
+
+        captured = self.capsys.readouterr()
+        self.assertEqual(
+            "trainable params: 294,912 || all params: 125,534,208 || trainable%: 0.23492560689115113\n", captured.out
```

Comment:
Instead of checking the exact message here, we could just check that this message is identical to the first one, i.e. that 4bit does not change the parameter count. That way, the test would be less dependent on the exact model being used.